### ISSUE TITLE
[Flaky Test]: `TestBeatsServerless/TestRunAndCheckData ` – elastic-agent logged an unexpected error: Error fetching data for metricset system.process

### ISSUE THREAD

**Issue by ycombinator** [elastic/elastic-agent#6431]
### Failing test case

TestBeatsServerless/TestRunAndCheckData

### Error message

elastic-agent logged an unexpected error: Error fetching data for metricset system.process: non fatal error; reporting partial metrics: error fetching PID metrics for 1 processes, most likely a "permission denied" error. Enable debug logging to determine the exact cause.

### Build

https://buildkite.com/elastic/elastic-agent-extended-testing/builds/5776#0193f478-fb7b-4f04-b543-140fe8b4a56b

### OS

Linux

### Stacktrace and notes

```markdown
>>> (linux-amd64-ubuntu-2404-default) Test output (sudo) (stdout): === FAIL: testing/integration TestBeatsServerless/TestRunAndCheckData (80.50s)
>>> (linux-amd64-ubuntu-2404-default) Test output (sudo) (stdout): beats_serverless_test.go:197:
>>> (linux-amd64-ubuntu-2404-default) Test output (sudo) (stdout): Error Trace:	/home/ubuntu/agent/testing/integration/beats_serverless_test.go:197
>>> (linux-amd64-ubuntu-2404-default) Test output (sudo) (stdout): Error:      	Received unexpected error:
>>> (linux-amd64-ubuntu-2404-default) Test output (sudo) (stdout): elastic-agent logged an unexpected error: Error fetching data for metricset system.process: non fatal error; reporting partial metrics: error fetching PID metrics for 1 processes, most likely a "permission denied" error. Enable debug logging to determine the exact cause.
>>> (linux-amd64-ubuntu-2404-default) Test output (sudo) (stdout): Test:       	TestBeatsServerless/TestRunAndCheckData
```

**Comment by ycombinator** [comment]
Another failure: https://buildkite.com/elastic/elastic-agent-extended-testing/builds/5759#0193f0d3-125d-4eec-b350-eb7236489ad2

**Comment by ycombinator** [comment]
Another failure: https://buildkite.com/elastic/elastic-agent-extended-testing/builds/5729#0193e160-c93c-412b-b4c5-5e3fd50db6fc

**Comment by ycombinator** [comment]
Looks like the failures started happening consistently with https://github.com/elastic/elastic-agent/commit/6062cb905054fe353cf20863feed10c655640781.  I don't think the failures are a result of the changes in that commit; more likely they are a result of a change in Beats.  The commit was merged into `main` on December 19, 2024 at 11:20 PM UTC.

**Comment by ycombinator** [comment]
This is the error message that's seen in the failing test:

> Error fetching data for metricset system.process: non fatal error; reporting partial metrics: error fetching PID metrics for 1 processes, most likely a "permission denied" error. Enable debug logging to determine the exact cause.

It looks like this message was introduced on December 12, 2024 ago via https://github.com/elastic/elastic-agent-system-metrics/pull/195.  A new release of `elastic-agent-system-metrics`, [0.11.5](https://github.com/elastic/elastic-agent-system-metrics/releases/tag/v0.11.5), containing the above change (and other changes as well), was made on the same day.  This new release was [pulled into Beats](https://github.com/elastic/beats/pull/41965) on the same day as well.

@VihasMakwana you are probably most familiar with this area of code so I'm assigning this issue to you for investigation.  Thanks.

**Comment by ycombinator** [comment]
Another failure: https://buildkite.com/elastic/elastic-agent-extended-testing/builds/5643#0193dcda-eafb-402e-9506-a5c073ea60b7

**Comment by ycombinator** [comment]
@VihasMakwana I think we can close this issue now that you have resolved the underlying cause of the test flakiness?

**Comment by VihasMakwana** [comment]
Closed by https://github.com/elastic/elastic-agent/pull/6438

**Comment by archon810** [comment]
What version is the fix available in? We're currently seeing this error in our production logs on version 8.17.1.

**Comment by mauri870** [comment]
> What version is the fix available in?

The fix should be available in 8.16.3 and 8.17.1.
